src/components/DocExplorer/Argument.js

```diff
@@ -0,0 +1,35 @@
+import React, { PropTypes } from 'react';
+import TypeLink from './TypeLink';
+
+const Argument = ({ arg, onClickType }) => {
```

Comment:
I think that it could be done in any way, but it's important to maintain consistency through the project.

When I refactored the component to this state:

```js
export default class Argument extends React.Component {
  static propTypes = {
    arg: PropTypes.object.isRequired,
    onClickType: PropTypes.func.isRequired,
  }

  render() {
    const arg = this.props.arg;

    return (
      <span className="arg">
        <span className="arg-name">{arg.name}</span>
        {': '}
        <TypeLink type={arg.type} onClick={this.props.onClickType} />
        {
          arg.defaultValue !== undefined &&
          <span>
            {' = '}
            <span className="arg-default-value">
              { arg.defaultValue === '' ? '""' : arg.defaultValue }
            </span>
          </span>
        }
      </span>
    );
  }
}
```

the linter was showing that it should be done in a pure function.

![screen shot 2017-01-17 at 14 30 43](https://cloud.githubusercontent.com/assets/4459232/22022283/9d308cba-dcc1-11e6-88a9-dc59eefac463.png)

But then I realize that I was probably missing the `shouldComponentUpdate` method. After that it was green.